### ISSUE TITLE
index step with FAST5 in multiple subdirectories

### ISSUE THREAD

**Issue by davidfarr** [jts/nanopolish#484]
Not so much an "issue" as a question! Let me know if there's a better way to ask...

We had MinKNOW do base calling for us, and did long reads with the MinION. The output of that process gave us a number of folders for every run, and then while there was only one directory within every run that contained FASTQ, there would be up to 40 folders under the pass folder containing hundreds of FAST5 files.

Originally I just ran the whole pipeline for calling methylation by directory, which worked, but I was wondering if there is a way I can pair the new concatenated FASTQ files per genetic sample with a way to pass an argument or in-line execution of a find command (bash) such that instead of dragging all the FAST5 files into one folder, I can "tell" the index command for nanopolish to index all the FAST5 files in each of those subdirectories and then from then on, just use the concatenated FASTQ file, the reference .FA and the index to call methylation and methylation frequency. I'm a bit of a novice at bash so there may or may not be an easy answer.

Right now if I were to throw all the FAST5 files into one hole, this is the command that should execute:
`${nanopolish} index -d /Volumes/data/reads/DD_combined/fast5_repository/ /Volumes/data/reads/DD_combined/merged/DD_merged.fq`

**Comment by jts** [comment]
There are a few ways to do this. The first way is to pass in multiple directories to index like this:

```
nanopolish index -d dir1 -d dir2 -d dir3 dd_merged.fq
```

The second way is to make a directory containing symlinks to the individual run folders. Something like:

```
mkdir all_runs
cd all_runs
ln -s /path/to/dir1
ln -s /path/to/dir2
ln -s /path/to/dir3
cd ..
nanopolish index -d all_runs merged.fq
```

The third way is to copy or symlink all of the individual fast5s into one directory. This is probably not advised since opening up huge directories can be slow.

I hope that helps!
Jared

**Comment by lijingjing1** [comment]
In my experience nanopolish index step will cost a few days to generate index.readdb,maybe you can generate a index.readdb txt by hands with the sequencing_summary.txt just sort the read_id and filename.It will be much more faster.

**Comment by jts** [comment]
There is an option to take in the sequencing summary file (-s), which will
make nanopolish index much faster.

Jared

**Comment by davidfarr** [comment]
Sorry for the delay responding to this - I had a hardware failure and connection issues that slowed me down. Such is life! I was able to get this working. I'm not sure how to approach the suggestion on generating the index.readdb by doing handwork, but I generated symlinks for every fast5 folder (there were about 80 links to make) in the sample and putting them into one folder. The index process seemed to be able to follow each of those links and I've progressed to using minmap2 and samtools for sorting. I'll update if anything goes wrong, but for now I feel comfortable closing the issue and thanks for your help!
